docs/xml/metainfo-component.xml

```diff
@@ -834,6 +834,78 @@
 				</listitem>
 			</varlistentry>
 
+			<varlistentry id="tag-requires-recommends-display-width-height">
+				<term>&lt;display_width/&gt; &amp; &lt;display_height/&gt;</term>
+				<listitem>
+				<para>
+					Set a relation to the display width and/or height defined as an integer value in either
+					<emphasis>millimeters</emphasis> or <emphasis>logical pixels</emphasis> (pixels divided by scaling factor).
+					To use millimeters, the <literal>unit</literal> property of this tag needs to be set to value <code>mm</code>, to use logical pixels
+					its value must be <code>px</code>. If no unit is explicitly defined, <code>mm</code> is implicitly assumed.
+				</para>
+				<para>
+					Relations of the display size can be defined using a <literal>compare</literal> property as described for <literal>requires</literal>/<literal>recommends</literal>.
+					If this property is not present, a value of <code>ge</code> (greater-or-equal) is implicitly assumed.
+				</para>
+				<para>
+					The <literal>display_width</literal> and <literal>display_height</literal> tags also accept one of the following text values,
+					which are implementation-defined and should roughly match the device class listed next to them:
+				</para>
+				<itemizedlist>
```

Comment:
I would avoid baking subjective measures into the stadard. I do understand that people want the simplest possible solution, but I'd stick to objective measures. Subjective definitions will change over time, causing bad user experience, e.g. "why are all applications too big for my 7 year old phone?" because the "medium" tablet is now better served by the "small" category, or "half of the applications don't fit" when someone makes a small phone which is 10cm in height but the software center accepts both "xsmall" and "small" because neither is really defined.

If anything, I would rather disseminate templates for those sizes, made up of objective measures, for copy and pasting by packagers.

Reply:
I think I agree, seeing as things like minimum size need to be precise values that developers actually test against. I don't think it makes sense for apps to subscribe to something that might change from under them, because the real problem is making apps work at new sizes, not the definition of "small" changing over time.

Reply:
> things like minimum size need to be precise values that developers actually test against

I don't think so - the point of the placeholder values is to not give precise measurements, as software centers don't need this. Of course, apps need precise pixel dimensions, but the get that info from the compositor at runtime.
The point of placeholders is to just give app developers a possibility to say "this app roughly runs on large/small/tiny screens" and leave it up to the software center to actually judge whether the device's screen fits that model.
This should avoid situations where a device is just a mm/pixel too small to show a particular app, even though it would run fine. By giving absolute values (which some developers will absolutely do and have to do!) you are very strict about the sizes of displays that your app will be installable on.

@dcz-purism The device class of these sizes shouldn't change - `xsmall` will always be "smaller then phone", any "medium" size will always be around a 13'' laptop. If we need bigger sizes in future, we would have to add new named placeholders (tiny? xxlarge?). That definitely would need clarification in the text if we keep placeholders in the spec, so software centers don't use completely arbitrary values and redefine "medium" to suddenly mean "medium-sized phones".
In the long run I expect this to sort itself out - as apps like Blender add a "larger equal medium display size" requirement, software centers will not redefine "medium" to mean "medium phones" as otherwise Blender would show up on the tiny screen.
This certainly needs clarification though.

Reply:
This causes two problems: sizes are on a continuum, and it's easily predictable that intermediate and extreme sizes will appear. Settling on a small number of values and waiting until things break to add easily predictable missing ones doesn't sound like a good idea to me.

The other issue that subjective measures are as much defined as fiat, as by usage. What recourse do we have when copy-pasting the wrong value becomes commonplace? I'm not convinced that asking nicely is going to prevent drift either.

Reply:
Android does exactly this, allowing no absolute dimensions to be set - so I am fairly confident that this will work just as well for us.
Placeholder sizes would only be used by developers if they really do not care about exact dimensions, once they do they wouldn't make the change. Also, device vendors are expected to do sensible things, and once an established ecosystem exists, I am sure they will.

On the copy & paste matter: That has been a massive issue since basically forever, and is one of the reasons why the MetaInfo Creator exists: https://www.freedesktop.org/software/appstream/metainfocreator/#/
But in general, if people copy&paste the wrong values, it's their fault and a bug that they should fix. The AppStream lint tool will be able to find & fix any obvious cases, but for the rest, the developer has to watch out for it.

Reply:
> This should avoid situations where a device is just a mm/pixel too small to show a particular app, even though it would run fine.

Except that's not true. If the app needs one pixel more than the screen width, it does not run fine, and shouldn't be shown in software centers. I really don't think introducing these shorthands buys us anything from an app developer or software center point of view, but it has the potential to muddy the waters so we'll end up with unreliable metadata.